src/Storage/Imap.php

```diff
@@ -465,20 +475,20 @@ public function renameFolder($oldName, $newName)
      *     folder is taken
      * @param null|array $flags set flags for new message, else a default set
      *     is used
+     * @param null|string $date set date for new message in the form of
+     *     "d-M-Y H:i:s O", else current time is used
      * @throws Exception\RuntimeException
      */
-    public function appendMessage($message, $folder = null, $flags = null)
+    public function appendMessage($message, $folder = null, $flags = null, $date = null)
```

Comment:
This is a potential BC break for anybody extending the class. I think the liklihood of a conflict is low (it would only be an issue if the argument is typehinted in the extension), but we'll need to message it in the release notes at the very least.